### ISSUE TITLE
add OpenRouter as supported endpoint for completions 

### ISSUE THREAD

**Issue by ShamanicArts** [arshad-yaseen/monacopilot#15]
Description:

I am requesting that Monacopilot add support for OpenRouter as an endpoint. OpenRouter provides access to a wide range of models beyond OpenAI and Anthropic, including open-source and community-provided models like DeepSeek Coder. This integration would allow Monacopilot users to utilize a broader selection of models with minimal changes.

Benefits:

Support for Open-Source Models: OpenRouter offers access to models from various providers, including open-source and community models that are becoming increasingly relevant for developers.

Broader Model Selection: By supporting OpenRouter, Monacopilot can offer users more flexibility and coverage for different use cases, including access to models outside of proprietary providers like OpenAI and Anthropic.

Minimal Overhead: OpenRouter is compatible with OpenAI’s API, meaning Monacopilot could support OpenRouter with minimal adjustments to existing functionality.

For more information on OpenRouter, please visit https://openrouter.ai/.

Example from OpenRouter Documentation:

Below is an example from the OpenRouter documentation showing how to make a completion request using their API:

```typescript

fetch("https://openrouter.ai/api/v1/chat/completions", {
  method: "POST",
  headers: {
    "Authorization": `Bearer ${OPENROUTER_API_KEY}`,
    "HTTP-Referer": `${YOUR_SITE_URL}`, // Optional, for including your app on openrouter.ai rankings.
    "X-Title": `${YOUR_SITE_NAME}`, // Optional. Shows in rankings on openrouter.ai.
    "Content-Type": "application/json"
  },
  body: JSON.stringify({
    "model": "openai/gpt-3.5-turbo",
    "messages": [
      {"role": "user", "content": "What is the meaning of life?"}
    ]
  })
});

```

Supporting OpenRouter would provide Monacopilot with access to a variety of models and a more flexible infrastructure for users who need different types of completions.

Thank you for considering this request.

**Comment by arshad-yaseen** [comment]
Hey @shamanicvocalarts, thank you for the suggestion! I'll be adding support for connecting any endpoint or AI service to Monacopilot soon, not just OpenRouter. With this, you will be able to connect OpenRouter, so count on it!

**Comment by arshad-yaseen** [comment]
We have released `v0.9.29` 🎉! Please update and do this for your case:

```javascript
const copilot = new Copilot(process.env.OPENROUTER_API_KEY, {
  model: {
    config: (apiKey, prompt) => ({
      endpoint: 'https://openrouter.ai/api/v1/chat/completions',
      headers: {
        'Content-Type': 'application/json',
        Authorization: `Bearer ${apiKey}`,
      },
      body: {
        model: 'openai/gpt-3.5-turbo',
        messages: [
          {role: 'system', content: prompt.system},
          {role: 'user', content: prompt.user},
        ],
      },
    }),
    transformResponse: response => ({
      completion: response.choices[0].message.content,
    }),
  },
});
```

For more information, refer to the [Custom Model](https://github.com/arshad-yaseen/monacopilot?tab=readme-ov-file#custom-model) documentation.
